packages/lockfile-file/src/read.ts

```diff
@@ -32,24 +34,60 @@ export async function readWantedLockfileAndAutofixConflicts (
   opts: {
     wantedVersion?: number
     ignoreIncompatible: boolean
+    useGitBranchLockfile?: boolean
+    mergeGitBranchLockfiles?: boolean
   }
 ): Promise<{
     lockfile: Lockfile | null
     hadConflicts: boolean
   }> {
-  const lockfilePath = path.join(pkgPath, WANTED_LOCKFILE)
-  return _read(lockfilePath, pkgPath, { ...opts, autofixMergeConflicts: true })
+  const lockfileNames: string[] = [WANTED_LOCKFILE]
+  if (opts.useGitBranchLockfile) {
+    const gitBranchLockfileName: string = await getWantedLockfileName(opts)
+    if (gitBranchLockfileName !== WANTED_LOCKFILE) {
+      lockfileNames.unshift(gitBranchLockfileName)
+    }
+  }
```

Comment:
I am not sure I understand what happens in this function. If `useGitBranchLockfile` is true, should we return the lockfile here? Without merging it with anything?

Reply:
Answered below.

> The first time pnpm-lock.feat-a.yaml is missing. So the readWantedLockfile returns the content of pnpm-lock.yaml